app/multitenant/memcache_client.go

```diff
@@ -0,0 +1,131 @@
+package multitenant
+
+import (
+	"bytes"
+	"fmt"
+	"net"
+	"sort"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/bradfitz/gomemcache/memcache"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/weaveworks/scope/report"
+)
+
+var (
+	memcacheCounter = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: "scope",
+		Name:      "memcache",
+		Help:      "Reports that missed our in-memory cache but went to our memcache",
+	}, []string{"result"})
+
+	memcacheRequestDuration = prometheus.NewSummaryVec(prometheus.SummaryOpts{
+		Namespace: "scope",
+		Name:      "memcache_request_duration_nanoseconds",
+		Help:      "Time spent doing memcache requests.",
+	}, []string{"method", "status_code"})
+)
+
+func init() {
+	prometheus.MustRegister(memcacheCounter)
+	prometheus.MustRegister(memcacheRequestDuration)
+}
+
+type memcacheClient struct {
+	client     *memcache.Client
+	expiration int32
+}
+
+// Create a new MemcacheClient that gets its server list from SRV and updates
+// the server list on a regular basis.
+func NewMemcacheClient(host string, timeout time.Duration, service string, updateInterval time.Duration, expiration int32) (*memcacheClient, error) {
+	var servers memcache.ServerList
+	client := memcache.NewFromSelector(&servers)
+	client.Timeout = timeout
+	err := updateMemcacheServers(host, service, &servers)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Returning the error here to let code elsewhere decide policy on error handling. 

There's a comment in `dynamo_collector.go` that explains what's going on.